octodns_azure/__init__.py

```diff
@@ -596,6 +605,20 @@ def _check_zone(self, name, create=False):
             create_zone = self._dns_client.zones.create_or_update
             create_zone(self._resource_group, name, Zone(location='global'))
             self._azure_zones.add(name)
+
+            # we create the zone so we should now be able to get its root ns
+            # records, they should be one of only a couple things that exist at
+            # this point.
+            records = self._dns_client.record_sets.list_by_dns_zone
+            values = []
+            for azrecord in records(self._resource_group, name):
```

Comment:
Just realized there's a [`get` record method](https://docs.microsoft.com/en-us/python/api/azure-mgmt-dns/azure.mgmt.dns.v2018_05_01.operations.recordsetsoperations?view=azure-python#azure-mgmt-dns-v2018-05-01-operations-recordsetsoperations-get). We should use that instead of listing all records.

Reply:
I can work on this bit

Reply:
Fwiw right after creation there's only 2 records, the root NS and SOA.

Reply:
Yep, it's just cleaner code-wise.